File: ledger-api/src/main/java/com/ledger/entity/Budget.java
package com.ledger.entity;

import jakarta.persistence.*;
import lombok.*;

import java.math.BigDecimal;

@Entity
@Table(name = "budgets", uniqueConstraints = {
    @UniqueConstraint(columnNames = {"user_id", "category_id", "type"})
})
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Budget {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "category_id", nullable = false)
    private String categoryId;

    /** monthly | yearly */
    @Column(nullable = false, length = 16)
    private String type;

    @Column(nullable = false, precision = 12, scale = 2)
    private BigDecimal amount;

    @Column(name = "user_id")
    private String userId;

    @PrePersist
    @PreUpdate
    protected void onSave() {
        if (userId == null) userId = "default";
    }
}

File: ledger-app/src/api/budget.js
import { get, put } from '@/utils/request.js'

/** 获取预算（年 + 月） */
export function getBudget() {
  return get('/api/budget')
}

/** 更新年预算，传入 { categoryId: amount } */
export function updateYearlyBudget(yearly) {
  return put('/api/budget/yearly', yearly)
}
